ibis/pandas/aggcontext.py

```diff
@@ -360,6 +367,14 @@ def agg(self, grouped_data, function, *args, **kwargs):
                 assert isinstance(function, str)
                 method = operator.methodcaller(function, *args, **kwargs)
 
+            max_lookback = self.max_lookback
+            if max_lookback is not None:
+                agg_method = method
+
+                def f(s):
+                    return agg_method(s.iloc[-max_lookback:])
```

Comment:
is there a reason you are not using the native pandas methods when max_lookback is an integer (not sure if it can be an offset here). using apply is nice for exteme generaility, but going to be *very* slow for the common functions (min, max, sum, etc).

Reply:
max_lookback will always be an integer here, specifying the number of rows (from the RowsWithMaxLookback definition).

We could create a follow-up to optimize to use the native pandas aggregation methods if possible and keeping the generalized apply as a fallback?

Reply:
This particular piece of functionality requires that we potentially shave down the rows *per window*, which I don't think is possible with the native methods. @jreback Is there actually a way to express what we want using the native method?

Reply:
so IIUC, given a frame you are doing an integer rolling operation with a specific number of rows? or something else; maybe this is better discussed in meat space.